src/Compilers/CSharp/Test/Syntax/Parsing/AwaitParsingTests.cs

```diff
@@ -1913,6 +1914,87 @@ public void BadStatementInSyncContext()
             EOF();
         }
 
+        [WorkItem(25410, "https://github.com/dotnet/roslyn/issues/25410")]
+        [Fact]
+        public void AwaitIdentifierExpressionStatementInSyncContextWithinAsyncMethod()
+        {
+            UsingTree(@"
+class C
+{
+    async void M()
+    {
+        () => {
```

Comment:
❓ Why not `Action action = () => {`

Reply:
Because it's not necessary. I thought we should always create the smallest repro?

Reply:
@sharwell This is consistent with other tests in that file. For example here:
https://github.com/dotnet/roslyn/blob/0dd9b972d6f600b696f13765b2ea7f90be19bec2/src/Compilers/CSharp/Test/Syntax/Parsing/AwaitParsingTests.cs#L209-L224

Reply:
I prefer to not fix this unless you want me to change the other uses of this in that file as well. But it would require changing the tests to expect a larger syntax tree. Therefore I think it's unnecessary and not worth it. It's normal for parser tests to test broken code.

Reply:
💭 How to balance "correctness" with "minimal", especially considering that this test intends to test behavior on incorrect code (`await` inside a non-`async` method). I'll leave the decision to the compiler team thanks!